pkg/controllers/dynakube/extension/reconciler_test.go

```diff
@@ -0,0 +1,94 @@
+package extension
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/Dynatrace/dynatrace-operator/pkg/api/scheme/fake"
+	"github.com/Dynatrace/dynatrace-operator/pkg/api/v1beta3/dynakube"
+	"github.com/Dynatrace/dynatrace-operator/pkg/util/conditions"
+	testutil "github.com/Dynatrace/dynatrace-operator/pkg/util/testing"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	testName      = "test-name"
+	testNamespace = "test-namespace"
+)
+
+func TestReconciler_Reconcile(t *testing.T) {
+	t.Run(`Extension secret not generated when Prometheus is disabled`, func(t *testing.T) {
+		instance := makeTestDynakube()
+		instance.Spec.Extensions.Prometheus.Enabled = false
+
+		fakeClient := fake.NewClient()
+		r := NewReconciler(fakeClient, fakeClient, instance)
+		err := r.Reconcile(context.Background())
+		require.NoError(t, err)
+
+		// assert extensions token is not generated
+		var secretFound corev1.Secret
+		err = fakeClient.Get(context.Background(), client.ObjectKey{Name: testName + "-extensions-token", Namespace: testNamespace}, &secretFound)
+		require.True(t, k8serrors.IsNotFound(err))
+
+		// assert conditions are empty
+		require.Empty(t, instance.Conditions())
+	})
+	t.Run(`Extension secret is generated when Prometheus is disabled`, func(t *testing.T) {
+		instance := makeTestDynakube()
```

Comment:
test says: Prometheus is disabled but `makeTestDynakube` hardcode
```
Prometheus: dynakube.PrometheusSpec{
					Enabled: true,
				},
```

Reply:
True! it was a typo. [Fixed](https://github.com/Dynatrace/dynatrace-operator/pull/3485/commits/3def3f6a0eeb9547c6b6538281f4e8ce1e6569a9)